src/amo/pages/Feedback/index.js

```diff
@@ -0,0 +1,478 @@
+        ),
+      },
+      {
+        value: 'other',
+        label: i18n.gettext('Something else'),
+        help: i18n.gettext(
+          'Anything that does’t fit into the other categories',
+        ),
+      },
+    ],
+  };
+};
+
+export class FeedbackBase extends React.Component<InternalProps, State> {
+  static defaultProps: DefaultProps = {
+    _window: typeof window !== 'undefined' ? window : {},
+    _config: config,
+  };
+
+  constructor(props: InternalProps) {
+    super(props);
+
+    const { dispatch, errorHandler, currentUser } = props;
+
+    this.state = {
+      successMessage: null,
+      ...this.getFormValues(currentUser),
+    };
+
+    dispatch(setViewContext(VIEW_CONTEXT_HOME));
+
+    if (errorHandler.hasError()) {
+      log.warn('Not loading data because of an error.');
+    }
+  }
+
+  componentDidUpdate(prevProps: InternalProps, prevState: State) {
+    if (
+      (!prevProps.errorHandler.hasError() &&
+        this.props.errorHandler.hasError()) ||
+      (!prevState.successMessage && this.state.successMessage)
+    ) {
+      this.props._window.scroll(0, 0);
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.errorHandler.clear();
+  }
+
+  onFieldChange: (event: SyntheticEvent<HTMLInputElement>) => void = (
+    event: SyntheticEvent<HTMLInputElement>,
+  ) => {
+    // event.preventDefault();
+
+    const { name, value } = event.currentTarget;
+
+    this.setState({
+      [name]: value,
+      successMessage: null,
+    });
+  };
+
+  onSubmit: HTMLElementEventHandler = (event: ElementEvent) => {
+    event.preventDefault();
+
+    const { dispatch, errorHandler, addonIdentifier } = this.props;
+    const { email, name, text, category } = this.state;
+
+    invariant(text.trim().length, 'A report cannot be sent with no content.');
+    invariant(addonIdentifier, 'An add-on identifer is needed for a report.');
+
+    dispatch(
+      sendAddonAbuseReport({
+        errorHandlerId: errorHandler.id,
+        addonSlug: addonIdentifier,
+        reporter_email: email,
+        reporter_name: name,
+        message: text,
+        reason: category,
+      }),
+    );
+  };
+
+  getFormValues(currentUser: UserType | null): FormValues {
+    const defaultFormValues = {
+      email: '',
+      name: '',
+      text: '',
+      category: null,
+    };
+
+    const { email, display_name: name } = currentUser || {};
+
+    return {
+      ...defaultFormValues,
+      email: email || '',
+      name: name || '',
+    };
+  }
+
+  preventSubmit(): boolean {
+    const { loading } = this.props;
+    const { text, category } = this.state;
+
+    return loading || !category || !text || (text && text.trim() === '');
+  }
+
+  render(): React.Node {
+    const {
+      _config,
+      errorHandler,
+      i18n,
+      loading,
+      currentUser,
+      addonIdentifier,
+      abuseReport,
+    } = this.props;
+
+    if (!_config.get('enableFeatureFeedbackForm')) {
+      return <NotFoundPage />;
+    }
```

Comment:
I think we could omit this if we only add the `<Route />` when the config flag is set.